cmake/ElektraConfig.cmake.in

```diff
@@ -46,3 +43,81 @@ set (Elektra_BUILD_TYPE @Elektra_BUILD_TYPE@)
 set (Elektra_SHARED @BUILD_SHARED@)
 set (Elektra_FULL @BUILD_FULL@)
 set (Elektra_STATIC @BUILD_STATIC@)
+
+# ~~~
+# Invoke the `kdb gen` during the build.
+# 
+# elektra_kdb_gen (template parent_key output_name
+#                  INPUT <plugin> <file>
+#                  OUTPUT_DIR <output_dir>
+#                  KDB <kdb>
+#                  PARAMS <params>...
+#                  OPTIONS <options>...)
+#
+# This will call `<kdb> gen -F <plugin>=<file> <options>... template parent_key output_name <params>...` in the
+# directory <output_dir>.
+# 
```

Comment:
Can you also give an example of how to execute it?

Reply:
I can, but does it really help that much, if I add e.g.:

```
elektra_kdb_gen (highlevel "/sw/org/app/#0/current" elektragen ni spec.ini)
```

That is all you need to do. CMake will then know which files are produced by `kdb gen` and call it, if one of the files is required by a target.

Reply:
I think even this example might help. A full usage example might help even more (how to compile the generated files).

Reply:
Full examples can be found in the examples directory.

Reply:
Then adding links to the examples directories is all what is needed.

Reply:
How should I link to the examples? With a Github link? I don't think the external examples show up on the homepage.

Reply:
The website automatically turns all internal links that are not in the website to github links.